core/src/oauth/connection.rs

```diff
@@ -0,0 +1,175 @@
+use crate::oauth::{providers::github::GithubConnectionProvider, store::OAuthStore};
+use crate::utils;
+use anyhow::Result;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Copy, Serialize, PartialEq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum ConnectionProvider {
+    Confluence,
+    Github,
+    GoogleDrive,
+    Intercom,
+    Notion,
+    Slack,
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq, Deserialize)]
+pub struct FinalizeResult {
+    code: String,
+    access_token: String,
+    access_token_expiry: Option<u64>,
+    refresh_token: Option<String>,
```

Comment:
Makes sense nvm me !

Reply:
The question is more open for `code` which we don't really need to store but I think it's best to store it and since this object will be used for an hypotehtical `cli` audit command, it should be accessible here IMHO.